homeassistant/components/device_tracker/__init__.py

```diff
@@ -420,18 +423,21 @@ def async_seen(self, host_name: str=None, location_name: str=None,
         self.last_seen = dt_util.utcnow()
         self.host_name = host_name
         self.location_name = location_name
-        self.gps_accuracy = gps_accuracy or 0
+
         if battery:
             self.battery = battery
         if attributes:
             self._attributes.update(attributes)
 
-        self.gps = None
-
         if gps is not None:
             try:
                 self.gps = float(gps[0]), float(gps[1])
+                self.gps_accuracy = gps_accuracy or 0
```

Comment:
(reposting as it was lost)

My thought was that we should reset self.gps_accuracy upon failure to parse gps (as it's no longer relevant), but only set it if we have valid coords.